sdk/src/Services/S3/Custom/Transfer/Internal/BufferedPartDataHandler.cs

```diff
@@ -257,77 +257,61 @@ public void Dispose()
-                // The MemoryStream starts at position 0 and can grow up to initialBufferSize
-                using (var memoryStream = new MemoryStream(partBuffer, 0, initialBufferSize, writable: true))
-                {
-                    _logger.DebugFormat("BufferedPartDataHandler: [Part {0}] Reading response stream into buffer",
-                        partNumber);
-
-                    // Use GetObjectResponse's stream copy logic which includes:
-                    // - Progress tracking with events
-                    // - Size validation (ContentLength vs bytes read)
-                    // - Buffered reading with proper chunk sizes
-                    await response.WriteResponseStreamAsync(
-                        memoryStream,
-                        null, // destination identifier (not needed for memory stream)
-                        _config.BufferSize,
-                        cancellationToken,
-                        validateSize: true)
-                        .ConfigureAwait(false);
-                    
-                    int totalRead = (int)memoryStream.Position;
-                    
-                    _logger.DebugFormat("BufferedPartDataHandler: [Part {0}] Read {1} bytes from response stream",
-                        partNumber, totalRead);
+                _logger.DebugFormat("BufferedPartDataHandler: [Part {0}] Reading response stream into chunked buffers",
+                    partNumber);
 
-                    // Set the length to reflect actual bytes read
-                    downloadedPart.SetLength(totalRead);
+                // Write response stream to chunked buffer stream
+                // ChunkedBufferStream automatically allocates chunks as needed
+                await response.WriteResponseStreamAsync(
+                    chunkedStream,
+                    null,
+                    _config.BufferSize,
+                    cancellationToken,
+                    validateSize: true)
+                    .ConfigureAwait(false);
+                
+                _logger.DebugFormat("BufferedPartDataHandler: [Part {0}] Buffered {1} bytes into chunked stream",
+                    partNumber, chunkedStream.Length);
 
-                    if (totalRead != expectedBytes)
-                    {
-                        _logger.Error(null, "BufferedPartDataHandler: [Part {0}] Size mismatch - Expected {1} bytes, read {2} bytes",
-                            partNumber, expectedBytes, totalRead);
-                    }
+                if (chunkedStream.Length != expectedBytes)
+                {
+                    _logger.Error(null, "BufferedPartDataHandler: [Part {0}] Size mismatch - Expected {1} bytes, read {2} bytes",
+                        partNumber, expectedBytes, chunkedStream.Length);
                 }
                 
-                return downloadedPart;
+                // Switch to read mode and wrap in IPartDataSource
+                chunkedStream.SwitchToReadMode();
```

Comment:
is there a way to do this automatically? without needing to call it from here

Reply:
updated it to be automatic